src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -123,6 +124,18 @@ def load_arguments(self, _):
         c.argument('sku', options_list=['--sku', '-s'], help='policy sku.', arg_type=get_enum_type(['free', 'standard']))
         c.argument('notscopes', options_list='--not-scopes', nargs='+')
 
+    with self.argument_context('policy assignment create', resource_type=ResourceType.MGMT_RESOURCE_POLICY, min_api='2018-05-01') as c:
+        c.argument('location', arg_type=get_location_type(self.cli_ctx))
+
+    with self.argument_context('policy assignment create', resource_type=ResourceType.MGMT_RESOURCE_POLICY, arg_group='Managed Identity', min_api='2018-05-01') as c:
+        c.argument('assign_identity', nargs='*', validator=validate_msi, help="Assigns a system assigned identity to the policy assignment.")
+        c.argument('identity_scope', help="Scope that the system assigned identity can access")
+        c.argument('identity_role', options_list=['--role', '--identity-role'], help="Role name or id that will be assigned to the managed identity")
+
+    with self.argument_context('policy assignment identity', resource_type=ResourceType.MGMT_RESOURCE_POLICY, min_api='2018-05-01') as c:
+        c.argument('identity_scope', help="Scope that the system assigned identity can access")
+        c.argument('identity_role', options_list=['--role', '--identity-role'], help="Role name or id that will be assigned to the managed identity")
+
```

Comment:
Instead of repeating this, you can create a `CLIArgumentType('identity_role', …)` once with the metadata, and then just plug it into both of these scopes. Otherwise it boosts your maintenance burden moving forward.